src/xform_to_datalog/manifest_datalog_facts.cc

```diff
@@ -55,6 +55,11 @@ ManifestDatalogFacts ManifestDatalogFacts::CreateFromManifestProto(
   uint64_t recipe_num = 0;
   for (const arcs::RecipeProto &recipe_proto : manifest_proto.recipes()) {
     absl::flat_hash_map<std::string, uint64_t> particle_numbers;
+    absl::flat_hash_map<std::string, std::string> handle_ids;
+    for (const arcs::HandleProto &handle_proto : recipe_proto.handles()) {
+      handle_ids[handle_proto.name()] =
+          handle_proto.id().empty() ? handle_proto.name() : handle_proto.id();
```

Comment:
The behavior on `operator[]` for C++ maps is to get the mutable reference to the value mapped to the provided key, creating a new entry if the key was not previously present. Is that the behavior we want here? Usually, if we are relying upon the keys being unique in advance (not uniqued by their insertion into the map) I think it's better to use `insert` and `CHECK` that the returned pair's `second` field is `true`, thereby ensuring that a fresh insertion always occurs.

Reply:
Done.